### ISSUE TITLE
Sort tabyl columns on adorn_totals(c("row")) descending (highest to lowest; left to right)

### ISSUE THREAD

**Issue by m31uk3** [sfirke/janitor#487]
## Brief description of the problem

This may just be inexperience but my use of the tabyl function is frequent and I find myself always wanting to have the columns sorted based on the adorn_total values for each column.

```r
diamonds %>% 
  tabyl(clarity, cut) %>% 
  adorn_totals(c("col")) %>%
  arrange(desc(as.numeric(Total))) %>%
  adorn_totals(c("row")) %>%
  #### {INSERT logic to sort columns by adorn_total column "Total"} ####
  adorn_title()
```

**Comment by jzadra** [comment]
The best way I've found to do this is, as is often the case, to go to long and then back to wide:

```
arrange_cols <- function(.data, row_name) {
  row_name <- sym(row_name)
  
  .data %>% 
    pivot_longer(-!!row_name) %>% 
    mutate(!!row_name := recode(!!row_name, `Total` = "!Total")) %>% 
    group_by(!!row_name) %>% 
    arrange(!!row_name, value) %>% 
    pivot_wider() %>% 
    mutate(!!row_name := recode(!!row_name, `!Total` = "zTotal")) %>% 
    arrange(!!row_name) %>% 
    mutate(!!row_name := recode(!!row_name, `zTotal` = "Total"))
}


diamonds %>% 
  tabyl(clarity, cut) %>% 
  adorn_totals(c("col")) %>%
  arrange(desc(as.numeric(Total))) %>%
  adorn_totals(c("row")) %>% 
  arrange_cols("clarity")
```

If we want to add this to `janitor`, we should add a bit more error checking and also allow bare column names in the function call.

**Comment by sfirke** [comment]
This will be best solved outside of janitor.  Here's one I came up based on the first StackOverflow hit from my search.  I had to modify that one since in a tabyl, the left-most column is descriptive and should stay on the left.  With your example:
```
x <- diamonds %>% 
  tabyl(clarity, cut) %>% 
  adorn_totals(c("col")) %>%
  arrange(desc(as.numeric(Total))) %>%
  adorn_totals(c("row"))


x[, c(1, order(-x[nrow(x),-1]) + 1)] # the 1s and -1 is to work around the first column being character
```

The first five StackOverflow hits from this search appeared promising if you want to look into it more: https://www.google.com/search?q=order+columns+based+on+row+values+r
